src/ResourceManager/Insights/Commands.Insights/Metrics/GetAzureRmMetricDefinitionCommand.cs

```diff
@@ -44,28 +40,16 @@ public class GetAzureRmMetricDefinitionCommand : MonitorClientCmdletBase
         public string[] MetricName { get; set; }
 
         /// <summary>
-        /// Gets or sets the detailedoutput parameter of the cmdlet
+        /// Gets or sets the metricnamespace parameter of the cmdlet
         /// </summary>
-        [Parameter(ValueFromPipelineByPropertyName = true, HelpMessage = "Return object with all the details of the records (the default is to return only some attributes, i.e. no detail)")]
-        public SwitchParameter DetailedOutput { get; set; }
+        [Parameter(ValueFromPipelineByPropertyName = true, HelpMessage = "The metric namespace to query metric definitions for")]
+        public string MetricNamespace { get; set; }
 
         /// <summary>
-        /// Process the general parameters (i.e. defined in this class) and the particular parameters (i.e. the parameters added by the descendants of this class).
+        /// Gets or sets the detailedoutput parameter of the cmdlet
         /// </summary>
-        /// <returns>The final query filter to be used by the cmdlet</returns>
-        protected string ProcessParameters()
-        {
-            var buffer = new StringBuilder();
-            if (this.MetricName != null)
-            {
-                var metrics = this.MetricName
-                    .Select(n => string.Concat("name.value eq '", n, "'"))
-                    .Aggregate((a, b) => string.Concat(a, " or ", b));
-                buffer.Append(metrics);
-            }
-
-            return buffer.ToString().Trim();
-        }
+        [Parameter(ValueFromPipelineByPropertyName = true, HelpMessage = "Return object with all the details of the records (the default is to return only some attributes, i.e. no detail)")]
```

Comment:
@pkamath13 @gucalder nit: you don't need `ValueFromPipelineByPropertyName` for a `SwitchParameter`

Reply:
Removed.